Some more cleanup/modernization (#1877)
enkidevs/curriculum: javascript/core/flow-control/loops-part-2.md

---
## Content

> `for..in` loops allow you to iterate through an objects keys. E.g. the keys below are `firstName` and `lastName`:

Reviewer: ```suggestion
The `for..in` loop allows you to iterate through an object's keys. Consider the following `firstName` and `lastName` keys:
```

```javascript
let obj = {
   firstName: "John",
   lastName: "Smith"
};
```

The keys can be used to get the value:

```javascript
for (let key in obj) {
   console.log(key, obj[key]);
}

// "firstName" "John"
// "lastName" "Smith"
```

> **For of**

Reviewer: ```suggestion
### `for...of`
```

> `for..of` loops are used for looping through iterable objects such as arrays:

Reviewer: ```suggestion
The `for..of` loops are used for looping through iterable objects such as arrays:
```

```javascript
let arr = [10, 20, 30];

for (let i of arr) {

>    console.log(i); //10 20 30

Reviewer: ```suggestion
   console.log(i); 
   // 10 20 30
```

}
```

> **Why use for of?**

Reviewer 2: Suggestion:
"**Why use for of?**" -> "**Why use `for of`?** This can help non-fluent English users.

Reviewer 2: Same can be applied to the "**For of**" on line `42`

Author: ```suggestion
**Why use `for..of`?**
```

> **Why use `for..of`?**

Reviewer: ```suggestion
### Why use `for..of`?
```

`for..of` loops will only iterate over an object's values and **not** its inherited properties like the `for..in` loop will.

Let's see an example. 

Notice how the `for..in` loop below returns `newFunc` that was added to array's prototype (we'll look at prototypes shortly but for now think of it as a way to add functionality):

```javascript
Array.prototype.newFunc = function() {};

let arr = [10, 20, 30];

for (let i in arr) {
   console.log(i); 
}
// 10 20 30 newFunc

for (let i of iterable) {
   console.log(i);
}
// 10 20 30

```

> **break and continue**

Reviewer: ```suggestion
### `break` & `continue`
```

> The `break` statement exits out of a loop and the `continue` statement will skip to the next iteration:

Reviewer: ```suggestion
The `break` statement exits out of a loop, while the `continue` statement will skip to the next iteration:
```

To exit a loop, you can use the ??? keyword.

* `break`
* `stop`
* `finish`

> * `end`

Reviewer: the answers already display as if enclosed by backticks,so we don't need to enclose in backticks anymore

![image](https://user-images.githubusercontent.com/19802557/77747332-6687f900-7016-11ea-9b77-8b17946a7980.png)

```suggestion
* break
* stop
* finish
* end
```
